client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx

```diff
@@ -13,24 +14,46 @@ import { Questions } from './questions';
 import { Sidebar } from './sidebar';
 import { SitePreview } from './site-preview';
 import { Steps } from './steps';
+import { useSteps } from './steps/use-steps';
 import type { Step } from '../../types';
 import './style.scss';
 
-const SiteMigrationInstructions: Step = function () {
+const SiteMigrationInstructions: Step = function ( { navigation } ) {
 	const site = useSite();
 	const siteId = site?.ID;
 	const { deleteMigrationSticker } = useMigrationStickerMutation();
 
+	useEffect( () => {
+		if ( siteId ) {
+			deleteMigrationSticker( siteId );
+		}
+	}, [ deleteMigrationSticker, siteId ] );
+
 	const queryParams = useQuery();
 	const importSiteQueryParam = queryParams.get( 'from' ) ?? '';
 	const { data: hostingDetails } = useHostingProviderUrlDetails( importSiteQueryParam );
 	const showHostingBadge = ! hostingDetails.is_unknown && ! hostingDetails.is_a8c;
 
 	const { detailedStatus } = usePrepareSiteForMigration( siteId );
 
+	const onComplete = () => {
+		navigation.submit?.( { destination: 'migration-started' } );
+	};
+
+	const { steps, completedSteps } = useSteps( { fromUrl: importSiteQueryParam, onComplete } );
+
 	const sidebar = (
-		<Sidebar>
-			<Steps fromUrl={ importSiteQueryParam } />
+		<Sidebar
```

Comment:
This approach reminded me of an article I came across last year. Though I'm not at all sure this falls under the same category, but just in case it helps you give it another check and be more sure p7H4VZ-4t0-p2

Reply:
Thank you for the reference here, Imran! But I think it's not the case here since it will always render.
You can think it similar to the `children`.

I just didn't use that in the `children` because it needs to be rendered in a different position of the `Sidebar`, which is a very simple component that only contains the structure to receive the other parts. :)